components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -1188,4 +1192,34 @@ protected void unsetRealmService(RealmService realmService) {
 
         OAuth2ServiceComponentHolder.getInstance().setRealmService(null);
     }
+
+    /**
+     * Set API Resource manager service.
+     *
+     * @param apiResourceManager API Resource Manager.
+     */
+    @Reference(
+            name = "api.resource.mgt.service.component",
+            service = APIResourceManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetAPIResourceManager")
+    protected void setAPIResourceManager(APIResourceManager apiResourceManager) {
+
+        log.debug("Setting the API Resource Manager.");
```

Comment:
move the debug log after actual setter method, change the log to indicate "Set the API resource manager in to `<xyz>` bundle"

Do the same in unset method

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2196/commits/1f6b9ab57f76c33caa73487e50b8cab7a2c3a317